### DISCUSSION TITLE
v.2.2.100

### DISCUSSION THREAD

**Discussion by Medard22** [Medard22/Dolphin-MMJR2-VBI#109] [Announcements]
This is a minor update with some handy fixes.

Changelog for v2.2.100

- Improvements to logic op approximation
- Removed obsolete "Skip Approximate Logic Op" Hack
- Fixed Touch IR Pointer mode selection (Thanks @ajay9634 for letting me know!)
- Some minor Android specific fixes
- MMJR2 updater is now hidden from the UI - as we all know, it wasn´t working anyway.
- Minor code updates

This release codebase is roughly equivalent to the Official Dolphin dev build 5.0-18515

**Savestates from v2.2.000 are compatible.** You should be able to update without any problems or revert back to v2.2.000 if you have issues.

<hr /><em>This discussion was created from the release <a href='https://github.com/Medard22/Dolphin-MMJR2-VBI/releases/tag/v2.2.100'>v.2.2.100</a>.</em>

**Comment by Nycolaz1** [comment]
I tested the main ones on the Wii (Smash, Kirby, Dragon Ball, Mario Kart and Resident Evil 4) and they all had a very high decrease in performance

**Reply by Medard22** [reply]
Hmm, there are no major changes to the pipeline for this release. The improved logic op approximation made changes in pixel shaders and shader cache. Other than that there was a minor update in Vulkan texture loader.

If you don't mind helping me we can bisect the issue. First of all, try to delete your shader cache and try again. If the performance is still worse let me know and I would send you a build with logic op approximation improvement only and we will test it. It really must be this or the changes in texture loader.

**Reply by Nycolaz1** [reply]
> Hmm, there are no major changes to the pipeline for this release. The improved logic op approximation made changes in pixel shaders and shader cache. Other than that there was a minor update in Vulkan texture loader.
> 
> If you don't mind helping me we can bisect the issue. First of all, try to delete your shader cache and try again. If the performance is still worse let me know and I would send you a build with logic op approximation improvement only and we will test it. It really must be this or the changes in texture loader.

I cleaned and rebuilt the shaders, but the frame rate actually dropped drastically in certain games (mario, smash and kirby)

**Reply by Medard22** [reply]
Ok here is a series of test builds. The first one is pretty much just v2.2.000 with improvements to Logic Op Approximations, the second one is the same as the previous one with changes to Vulkan texture loader and the third one is pretty much v2.2.100 with a one small hack related to movie player removed. It was introduced to Dolphin official and I noticed it spams some entries in the log. There is a very low chance this might the root cause of the problem, but let´s be thorough.

Now please test them in sequence and let me know where the performance drop happened. If it happens in the first one then you probably don´t have to test the rest. Links are from my personal Google Drive. Thank you for helping me out and sorry it took a while. I purged caches in my IDE between builds just to be sure.

[MMJR2-VBI-test01](https://drive.google.com/file/d/1i9qasLie1Kt3JUp4yIVg02_K25lWwphT/view?usp=drive_link)

[MMJR2-VBI-test02](https://drive.google.com/file/d/1jWKyuHdEOMNOW2xWQhPF-mQrNK_8BLxV/view?usp=drive_link)

[MMJR2-VBI-test03](https://drive.google.com/file/d/1qpECxtpfalbhUP71fAaAOHJk0M1SU3zW/view?usp=drive_link)

**Reply by Nycolaz1** [reply]
From my tests that I did on each one with the same configuration (only Mario Kart and Smash) tests 2 and 3 were more stable, but I'm going to test Vulkan now

**Reply by Nycolaz1** [reply]
> Ok here is a series of test builds. The first one is pretty much just v2.2.000 with improvements to Logic Op Approximations, the second one is the same as the previous one with changes to Vulkan texture loader and the third one is pretty much v2.2.100 with a one small hack related to movie player removed. It was introduced to Dolphin official and I noticed it spams some entries in the log. There is a very low chance this might the root cause of the problem, but let´s be thorough.
> 
> Now please test them in sequence and let me know where the performance drop happened. If it happens in the first one then you probably don´t have to test the rest. Links are from my personal Google Drive. Thank you for helping me out and sorry it took a while. I purged caches in my IDE between builds just to be sure.
> 
> [MMJR2-VBI-test01](https://drive.google.com/file/d/1i9qasLie1Kt3JUp4yIVg02_K25lWwphT/view?usp=drive_link)
> 
> [MMJR2-VBI-test02](https://drive.google.com/file/d/1jWKyuHdEOMNOW2xWQhPF-mQrNK_8BLxV/view?usp=drive_link)
> 
> [MMJR2-VBI-test03](https://drive.google.com/file/d/1qpECxtpfalbhUP71fAaAOHJk0M1SU3zW/view?usp=drive_link)

I don't know what's different about the code in test version 2, but if you improve what's there, your project will be more impeccable

**Reply by Medard22** [reply]
Ok let's break this down. I need to clear everything up before we proceed.

Test build one is v2.2.000 with improvement to logic ops. We can assume these doesn't affect performance.

Test build two is the same as the first one with small changes to vulkan texture loader. Since you say this version works well, we can assume these changes are safe as well.

The third build is the whole v2.2.100 but with a single hack from Dolphin official reverted. If that version works fine we have our culprit.

The rest of the changes in the third build was mostly unrelated code syncs in Dolphinqt, debugger etc. These shouldn't affect the Android version at all. I made some updates to the build file, some fixes to Android UI and IR mode settings fix. These may also cause unforeseen issues.

So my question is, is there a noticeable performance drop between build 2 and 3? If the answer is yes, let me know. When I get back from work I will have a look at the code changes and provide some additional test builds to narrow it down. If the answer is no, then we have found the root cause and I will update the release.

Again, thanks for the testing. You helped me a lot.

**Reply by ajay9634** [reply]
I am getting some graphics problems in metroid prime 3 (mmjr2 vbi).
It looks like a flickering issue .
Mostly when I scan an object in game.

**Reply by Medard22** [reply]
> I am getting some graphics problems in metroid prime 3 (mmjr2 vbi).
> It looks like a flickering issue .
> Mostly when I scan an object in game.

Try the test builds above and let me know in which one this problem shows up.

**Reply by ajay9634** [reply]
> > I am getting some graphics problems in metroid prime 3 (mmjr2 vbi).
> > It looks like a flickering issue .
> > Mostly when I scan an object in game.
> 
> Try the test builds above and let me know in which one this problem shows up. 

I tested all of mmjr 2 apk , it's all the same. Only the latest official apk has no problem .

https://github.com/Medard22/Dolphin-MMJR2-VBI/assets/78233144/296df026-f439-461e-9fa3-38e73b3c05b7

**Reply by Medard22** [reply]
The latest official apk? You mean the v2.2.000? Can you check the ini of the Metroid game for Skip Approximate Logic Op entry?

**Reply by ajay9634** [reply]
I mean from here

https://dolphin-emu.org/download/

**Reply by ajay9634** [reply]
> I mean from here
> 
> https://dolphin-emu.org/download/

Also Far Cry Vengeance is perfect only on official apk but in mmjr 2 , it has graphics bug

**Reply by Medard22** [reply]
I am sorry, but this isn't something I can fix anytime soon . Dolphin Official contains at least a whole year worth of code updates and fixes. MMJR2 was and still is a performance build. The trade offs are occasional graphical bugs and glitches. Dolphin Official on the other hand focuses mostly on stability and emulation accuracy, at the cost of a slightly lower performance. The performance drop is less substantial on a more powerful devices, but for low-power Android SoC like T618 it often a difference between playable and unplayable state. If you want to discuss more, feel free to add a new thread in discussions. For now, let us focus on bugs introduced in the latest release of MMJR2.

**Reply by Nycolaz1** [reply]
> Ok let's break this down. I need to clear everything up before we proceed.
> 
> Test build one is v2.2.000 with improvement to logic ops. We can assume these doesn't affect performance.
> 
> Test build two is the same as the first one with small changes to vulkan texture loader. Since you say this version works well, we can assume these changes are safe as well.
> 
> The third build is the whole v2.2.100 but with a single hack from Dolphin official reverted. If that version works fine we have our culprit.
> 
> The rest of the changes in the third build was mostly unrelated code syncs in Dolphinqt, debugger etc. These shouldn't affect the Android version at all. I made some updates to the build file, some fixes to Android UI and IR mode settings fix. These may also cause unforeseen issues.
> 
> So my question is, is there a noticeable performance drop between build 2 and 3? If the answer is yes, let me know. When I get back from work I will have a look at the code changes and provide some additional test builds to narrow it down. If the answer is no, then we have found the root cause and I will update the release.
> 
> Again, thanks for the testing. You helped me a lot. 

the drop wasn't that sudden, something from 5 fps from 2 to 3, and thank you for making such a good project

**Reply by Medard22** [reply]
Thank you, I have a few leads now. I will let you know when I need some further testing.

**Reply by Nycolaz1** [reply]
> Thank you, I have a few leads now. I will let you know when I need some further testing. 

Ok😁

**Reply by Medard22** [reply]
Ok this build fixed one small error with Fileutil and I reverted some changes to build.gradle. Please give it a go and compare the performance to test build 02. Thanks.

[MMJR2-VBI-test04](https://drive.google.com/file/d/1gtnt7w9LYcdvv0_ncdMjYUgxvacHMkPt/view?usp=drive_link)

**Reply by Nycolaz1** [reply]
> Ok this build fixed one small error with Fileutil and I reverted some changes to build.gradle. Please give it a go and compare the performance to test build 02. Thanks.
> 
> [MMJR2-VBI-test04](https://drive.google.com/file/d/1gtnt7w9LYcdvv0_ncdMjYUgxvacHMkPt/view?usp=drive_link)

2 is better

**Reply by Medard22** [reply]
I can't reproduce it on my RP3+. Mario Kart Wii is actually a bit faster in this build than v2.2.000 and Kirby: Return to Dreamland is pretty much the same. I noticed that there was more pronounced shader compilation stutter on v2.2.100, but that seems to be fixed now. In all games I tested I can't see any noticeable drop in performance. At least within a margin of measurement error.
What kind of device you have? Retroid, phone?

**Reply by Nycolaz1** [reply]
> I can't reproduce it on my RP3+. Mario Kart Wii is actually a bit faster in this build than v2.2.000 and Kirby: Return to Dreamland is pretty much the same. I noticed that there was more pronounced shader compilation stutter on v2.2.100, but that seems to be fixed now. In all games I tested I can't see any noticeable drop in performance. At least within a margin of measurement error.
> What kind of device you have? Retroid, phone? 

I have Samsung M52(Snapdragon 778g)

**Reply by Medard22** [reply]
Ok, I posted this test apk on retroid discord server. Hopefully I will get some additional feedback soon. I tested Mario Kart Wii, Kirby, Prince of Persia, Wind Waker, Twilight Princess and Mario Sunshine with no noticeable performance drop on my devices. I went over all changes from test02 upwards that could affect performance and found none. I will check the modified debugger code for issues and we shall see if any other users report any performance regression.

**Reply by Nycolaz1** [reply]
> Ok, I posted this test apk on retroid discord server. Hopefully I will get some additional feedback soon. I tested Mario Kart Wii, Kirby, Prince of Persia, Wind Waker, Twilight Princess and Mario Sunshine with no noticeable performance drop on my devices. I went over all changes from test02 upwards that could affect performance and found none. I will check the modified debugger code for issues and we shall see if any other users report any performance regression. 

but I previously commented that my problem was solved with the version 2 code, I only commented that version 4 was worse in fps than version 2, but it was more stable

**Reply by Medard22** [reply]
The thing is v4 should not have any fps drops. The code changes added shouldn't affect performance, and I can't reproduce the fps drop on my devices. (T618 and SD855+).

**Reply by Nycolaz1** [reply]
> The thing is v4 should not have any fps drops. The code changes added shouldn't affect performance, and I can't reproduce the fps drop on my devices. (T618 and SD855+). 

Humm I will continue testing to see if there is a margin of error

**Reply by Medard22** [reply]
From the feedback I collected so far, no other users reported FPS drops. Right now I will focus on shader precompilation bug that happened between 2.1.010 and 2.2.000. If I manage to fix it I will make a two separate releases. 2.1.100-hotfix that will have a minimum subset of changes. Pretty much test02 with fixes to fileutil, cache cleaning, IR pointer fix, small changes to the UI and the shader precompilation fix. This build would be for you and it should have no fps drops, at least I hope so. I will send you an RC build for testing when it´s ready. The second release will be 2.1.110 for everyone else. If anyone has a performance drop, they will have  2.1.100-hotfix to fall back to.  I need to keep the codebase as close to dolphin official as possible to implement further changes. If I start skipping code changes in the core, things will go out of hand very quickly.

**Reply by Medard22** [reply]
Hello I´ve made and rc build for the hotfix release. This one is based on v2.2.000 with just the minimum subset of code changes that should not affect performance. Please check it on your device and let me know how is the performance compared to the test02 apk. Thanks.

[MMJR2-VBI-v221-hotfix-rc](https://drive.google.com/file/d/1ipCf6_9aghJRiavazMlBrvOOWxgVCxKh/view?usp=sharing)

**Reply by ajay9634** [reply]
> Hello I´ve made and rc build for the hotfix release. This one is based on v2.2.000 with just the minimum subset of code changes that should not affect performance. Please check it on your device and let me know how is the performance compared to the test02 apk. Thanks.
> 
> [MMJR2-VBI-v221-hotfix-rc](https://drive.google.com/file/d/1ipCf6_9aghJRiavazMlBrvOOWxgVCxKh/view?usp=sharing)


I can test this.
I think the Metroid prime 3 graphics bug is fixed in this version .
But minor fps low

**Reply by Medard22** [reply]
I think I finally found the code responsible for performance drops after extensive testing of various builds in MarioKart. I hope this is the last one to test. Please compare this build with v2.2.000. Thank you.

[MMJR2-VBI-test06](https://drive.google.com/file/d/1ZpZXgwz3n5CovuhRdL0jH8Rc9oNFDK3K/view?usp=drive_link)
